README.md

```diff
@@ -426,6 +431,44 @@ fmt.Println(output)
 
 Filters are operations on an existing pipe that also return a pipe, allowing you to chain filters indefinitely.
 
+## Basename
+
+`Basename()` reads a list of filenames from the pipe, one per line, and returns a pipe that contains the last element of each line.
+
+If a line is empty, `Basename()` will return a `.`
+
+Trailing slashes are removed.
+
+`Basename()` will also trim off any extension that you provide, if the filename has one. Pass in an empty string if you don't want to trim any extensions.
```

Comment:
I don't mind either way. I added it to `Basename` because it's one of the things `basename(1)` is traditionally used for. It's just a shame Golang doesn't support optional arguments :(

Reply:
In general my attitude to these things is that if the logical/sensible behaviour is the same as the equivalent Unix command, I'm happy, but if it isn't, I'll go with what's sensible rather than be compatible with Unix. Similarly, I won't do things just because Unix does them. Let's have Basename just do basenames, and TrimExt just do trimming.